### PR TITLE
Opti: avoid `_heap.size` reads

### PR DESCRIPTION
**Pull request by MathisGD**
Small optimization of `HeapOrdering`. See each commit as a separated optimization, we can keep each of them separately. My preferred one is the first one, the second and the first one adding a little bit of code.

See the results for `updateHeap` in `TestRandomHeapOrdering`:
Gas before: 21922
Gas after: 21811

### CODE REVIEW

File: contracts/HeapOrdering.sol [hunk 1]
```diff
@@ -234,20 +239,24 @@ library HeapOrdering {
...
         uint256 index = _heap.indexOf[_id];
         uint256 accountsLength = _heap.accounts.length;
 
         // Swap the last account and the account to remove, then pop it.
         swap(_heap, index, accountsLength - 1);
-        if (_heap.size == accountsLength) _heap.size--;
+        if (_size == accountsLength) {
+            _size--;
+            _heap.size = _size;
+        }
```
Comment (QGarchery) [inline contracts/HeapOrdering.sol:253]:
```suggestion
        if (_size == accountsLength) _heap.size = --_size;
```

**Review by QGarchery — APPROVED** [review]
Let's go !

File: contracts/HeapOrdering.sol [hunk 2]
```diff
@@ -244,12 +244,15 @@ library HeapOrdering {
 
         // Swap the last account and the account to remove, then pop it.
         swap(_heap, index, accountsLength - 1);
-        if (_size == accountsLength) _heap.size = _size - 1;
+        if (_size == accountsLength) {
+            _size--;
+            _heap.size = _size;
```
Comment (MerlinEgalite) [inline contracts/HeapOrdering.sol]: and if you try just: `_heap.size = _size--;`?

And in an unchecked?

**Review by MerlinEgalite — CHANGES_REQUESTED** [review]
Can you try it please? Else LGTM!

File: contracts/HeapOrdering.sol (hunk 2 above)
Comment (MathisGD) [reply contracts/HeapOrdering.sol]: Yes it has already been changed ;-)

Comment (MerlinEgalite) [reply contracts/HeapOrdering.sol]: wut? I'm not sure what happened 😅

**Review by MerlinEgalite — APPROVED** [review]
LGTM

File: contracts/HeapOrdering.sol (hunk 2 above)
Comment (MathisGD) [reply contracts/HeapOrdering.sol]: I think that you started a review then I changed it and then you submitted your review :-)

**Pull request merged**